specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,262 @@
+
+; attributes
+
+attributes #0 = { "entry_point" "required_qubits"="2" "required_results"="2" }
+```
+
+TODO: do we need additional restrictions for output recording? Like e.g. it
+needs to be a dedicated block, or output recording functions can only occur in
+the last block of the entry point?
+
+TODO: do we need a `__quantum__rt__initialize` and `__quantum__rt__finalize`?
+TODO: do measurements need to be at the end of the quantum computations, or
+should the backend take care of deferring them if necessary?
+
+TODO: a profile identifier and version number within the IR would be good to
+have (may be needed for correct usage of qubit and result pointers) -> look into
+custom target triples
+
+## Entry Point Definition
+
+The bitcode contains the definition of the LLVM function that should be invoked
+when the program is executed, referred to as entry point in the rest of this
+profile specification. The name of this function may be chosen freely, as long
+as it is a valid [global
+identifier](https://llvm.org/docs/LangRef.html#identifiers) by LLVM standard.
+
+The entry point may not take any parameters and must return an exit code in the
+form of a 64-bit integer. The exit code 0 must be used to indicate a successful
+execution of the program.
+
+### Attributes
+
+The following custom attributes must be attached to the entry point function:
+
+- An attribute named "EntryPoint" that does not
+- An attribute indicating the total number of qubits used by the program
+- An attribute indicating the total number of classical bits required throughout
+  the program to store measurement outcomes
+
+These attributes will show up as an [attribute
+group](https://releases.llvm.org/13.0.1/docs/LangRef.html#attrgrp) in the IR.
+Attribute groups are numbered such that they can be easily referenced by
+multiple function definitions or global variables. Arbitrary custom attributes
+may be optionally attached to any of the declared functions to convey additional
+information about that function. Consumers of Base Profile compliant programs
+should hence not rely on the numbering of the entry point attribute group, but
+instead look for function to which an attribute with the name `entry_point` is
+attached to determine which one to invoke when the program is launched.
+
+To indicate the total number of qubits required to execute the program, a custom
+attribute with the name `required_qubits` is defined and attached to the entry
+point. To indicate the number of registers/bits needed to store measurement
+results during program execution, a custom attribute with the name
+`required_results` is defined and attached to the entry point. The value of both
+of these attributes is the string representation of a 64-bit integer constant.
+
+TODO: why the string value, and not an integer? TODO: no double underscore guard
+for attributes likely makes sense
+
+### Function Body
+
+The function body consists of a single block named `entry`. This implies in
```

Comment:
We should not rely on block names, and I edited this. 
Re: why use an attribute to mark entry points opposed to require a naming convention: I find that naming conventions - even if it is just for the entry point - get pretty awkward pretty quickly. For one, it might be desirable to support multiple entry points (not as part of the Base Profile), and I see no significant downside to using an attribute. Do you see a reason to prefer naming convention over an attribute?